### ISSUE TITLE
Misleading error message with malformed turbofishes

### ISSUE THREAD

**Issue by atamis** [rust-lang/rust#54521]
I typoed the angle brackets when attempting turbofish.

```rust
fn main() {
    println!("{:?}", vec![1, 2, 3].into_iter().collect::<Vec<usize>>>());
}

```

([Playground](https://play.rust-lang.org/?gist=dd291f4a7c8e184f571f8d00f8dc839c&version=nightly&mode=debug&edition=2015))

I expected to see this happen: To get an error indicating that I had an extra closing
angle bracket after my turbofish, or that I had a mismatched delimiter.

Here is what happened instead:

Errors:

```
   Compiling playground v0.0.1 (/playground)
error: field expressions may not have generic arguments
 --> src/lib.rs:2:55
  |
2 |     println!("{:?}", vec![1, 2, 3].into_iter().collect::<Vec<usize>>>());
  |                                                       ^^^^^^^^^^^^^^

error: aborting due to previous error

error: Could not compile `playground`.

To learn more, run the command again with --verbose.

```

This error message refers to the case when you attempt to use turbofish on a struct field
rather than a method or function, like `struct.field::<usize>`. However, this error message
is confusing to users because, if you look at the end of the collect call, there _are_ parentheses,
and this _looks_ like a correctly formed method call, which obscures the real error.

**Comment by atamis** [comment]
For reference, this is how the compiler responds to the same typo elsewhere.

```rust
fn main() {
    println!("{:?}", Vec::<usize>>::new());
}

```

([Playground](https://play.rust-lang.org/?gist=c194cac89405ed870fa110b94c2a9973&version=nightly&mode=debug&edition=2015))

```
   Compiling playground v0.0.1 (/playground)
error[E0423]: expected value, found struct `Vec`
 --> src/main.rs:2:22
  |
2 |     println!("{:?}", Vec::<usize>>::new());
  |                      ^^^^^^^^^^^^ did you mean `Vec { /* fields */ }`?

error[E0425]: cannot find function `new` in the crate root
 --> src/main.rs:2:37
  |
2 |     println!("{:?}", Vec::<usize>>::new());
  |                                     ^^^ not found in the crate root
```

**Comment by m-cat** [comment]
Just ran into this. Weird/confusing/unhelpful error message -- the real problem is an extra delimiter as you said.

**Comment by estebank** [comment]
The opposite problem should also be handled, probably as part of a separate PR: `vec![1, 2, 3].into_iter().collect::<Vec<usize>()) `
